alws/routers/uploads.py

```diff
@@ -14,15 +22,66 @@
 )
 
 
+async def get_repo_product(
+    session: AsyncSession,
+    repository: str,
+):
+    if repository.endswith("br"):
+        build = (
+            (
+                await session.execute(
+                    select(Build).filter(
+                        Build.repos.any(Repository.name.ilike(f'%{repository}'))
+                    )
+                )
+            )
+            .scalars()
+            .first()
+        )
+        if build:
+            return (
+                (
+                    await session.execute(
+                        select(Product).filter(Product.team_id == build.team_id)
+                    )
+                )
+                .scalars()
+                .first()
+            )
+    else:
+        return (
+            (
+                await session.execute(
+                    select(Product).filter(
+                        Product.repositories.any(
+                            Repository.name.ilike(f'%{repository}')
+                        )
+                    )
+                )
+            )
+            .scalars()
+            .first()
+        )
+
+
 @router.post("/upload_repometada/")
 async def upload_repometada(
     modules: typing.Optional[UploadFile] = None,
     comps: typing.Optional[UploadFile] = None,
     repository: str = Form(...),
     session: AsyncSession = Depends(get_db),
+    user: User = Depends(get_current_user),
 ):
     msg = ""
     uploader = MetadataUploader(session, repository)
+    repo_product = await get_repo_product(session, repository)
+    if not repo_product:
+        print(f"couldn't find a product or build repository {repository}")
+        return {"error": f"couldn't find a product or build repository {repository}"}
+    if not can_perform(repo_product, user, actions.ReleaseToProduct.name):
```

Comment:
does it work for a product without loading relationships?

Reply:
I'm curious too

Reply:
In the end it didn't, added loading relationships in the previous commit